### PR TITLE
Feature/devise auth

### PR DESCRIPTION
**Pull request by eduardosancho**
# What was done in this PR by @Bondok6 & @eduardosancho 🎃 : 

## Added Features:
- [x] Install devise.
- [x] Can register a new user with name, email, and password.
- [x] Hashed password is stored in the database.
- [x] Send an email to confirm the email address (you can find it in the terminal where you're running the rails server).
- [x] Log In requires an email and password (your email needs to be confirmed, otherwise you won't be able to log in).
- [x] If a user forgets their password, they can reset it via a provided link in an email (also found in the terminal).
- [x] User can now be deleted even if it has children (using the command destroy).

## Modified Features:
- [x] Remove current_user method from ApplicationController, since devise provides a better one.
- [x] Root route now depends on if the user has been authenticated or not.

### CODE REVIEW

File: db/migrate/20220429051159_add_user_ref_to_posts.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class AddUserRefToPosts < ActiveRecord::Migration[7.0]
   def change
-    add_reference :posts, :user, null: false, foreign_key: true
+    add_reference :posts, :user, null: false, index: true
+    add_foreign_key :posts, :users, column: :author_id, on_delete: :cascade
```
Comment (andersonlebon) [inline db/migrate/20220429051159_add_user_ref_to_posts.rb]: image

- [ ] Kindly correct this migration as is throwing an error. :point_up: 
Check this example :point_down: 

```rb 
class AddUserRefToPosts < ActiveRecord::Migration[7.0]
  def change
    add_reference :posts, :author, null: false, foreign_key: { to_table: 'users' }
  end
end 
```

File: db/migrate/20220429051320_add_user_ref_to_comments.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class AddUserRefToComments < ActiveRecord::Migration[7.0]
   def change
-    add_reference :comments, :user, null: false, foreign_key: true
+    add_reference :comments, :user, null: false, index: true
+    add_foreign_key :comments, :users, column: :author_id, on_delete: :cascade
```
Comment (andersonlebon) [inline db/migrate/20220429051320_add_user_ref_to_comments.rb]: Same here 

image


- [x] Kindly correct this migration as is throwing an error. :point_up: 
Check this example :point_down: 

```rb 
class AddUserRefToComments < ActiveRecord::Migration[7.0]  def change
    add_reference :comments, :author, null: false, index: true, foreign_key: { to_table: 'users' }
  end
end 
```

File: db/migrate/20220429051355_add_user_ref_to_likes.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class AddUserRefToLikes < ActiveRecord::Migration[7.0]
   def change
-    add_reference :likes, :user, null: false, foreign_key: true
+    add_reference :likes, :user, null: false, index: true
+    add_foreign_key :likes, :users, column: :author_id, on_delete: :cascade
```
Comment (andersonlebon) [inline db/migrate/20220429051355_add_user_ref_to_likes.rb]: Same here 

image


- [x] Kindly correct this migration as is throwing an error. :point_up: 
Check this example :point_down: 

```rb 
class AddUserRefToLikes < ActiveRecord::Migration[7.0]
  def change
    add_reference :likes, :author, null: false, index: true, foreign_key: { to_table: 'users' }
  end
end
```

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Bondok6 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — APPROVED** [review]
Hi, @Bondok6,

Tasks for this project have been successfully implemented 💯.

### Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**Please, remember to tag me in your question so I can receive the notification.**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**